### PR TITLE
feat: Dry-run in Apply/Create

### PR DESCRIPTION
**Pull request by shubham-cmyk**
fix : https://github.com/kyverno/chainsaw/issues/407

### CODE REVIEW

File: pkg/runner/operations/apply.go [hunk 1]
```diff
@@ -41,27 +48,38 @@ func operationApply(ctx context.Context, obj ctrlclient.Object, c client.Client,
...
 				return false, err
 			} else if shouldFail {
 				return false, errors.New("an error was expected but didn't happen")
 			}
+			if dryRun {
+				logger.Log(operation, color.BoldYellow, "DRY RUN: Resource patch simulated")
+			}
 		} else if kerrors.IsNotFound(err) {
-			if err := c.Create(ctx, obj); err != nil {
+			createOptions := []ctrlclient.CreateOption{}
```
Comment (eddycharly) [inline pkg/runner/operations/apply.go:64]:
```suggestion
			var createOptions []ctrlclient.CreateOption
```

File: pkg/runner/operations/create.go [hunk 2]
```diff
@@ -26,25 +27,33 @@ func operationCreate(ctx context.Context, obj ctrlclient.Object, c client.Client
...
 				return false, err
 			} else {
-				if cleaner != nil {
+				if cleaner != nil && !dryRun {
 					cleaner(obj, c)
 				}
 				if shouldFail {
 					return false, errors.New("an error was expected but didn't happen")
 				}
+				if dryRun {
```
Comment (eddycharly) [inline pkg/runner/operations/create.go:54]: Do we need a dedicated log ?

File: pkg/apis/v1alpha1/apply.go [hunk 3]
```diff
@@ -6,6 +6,9 @@ type Apply struct {
 	// FileRefOrResource provides a reference to the file containing the resources to be applied.
 	FileRefOrResource `json:",inline"`
 
+	// DryRun determines whether the file should be applied in dry run mode.
+	DryRun bool `json:"dryRun,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/apply.go:10]: Should we make this a pointer ?

File: pkg/apis/v1alpha1/create.go [hunk 4]
```diff
@@ -6,6 +6,9 @@ type Create struct {
 	// FileRefOrResource provides a reference to the file containing the resources to be created.
 	FileRefOrResource `json:",inline"`
 
+	// DryRun determines whether the file should be applied in dry run mode.
+	DryRun bool `json:"dryRun,omitempty"`
```
Comment (eddycharly) [inline pkg/apis/v1alpha1/create.go:10]: Should we make this a pointer ?

File: pkg/apis/v1alpha1/apply.go (hunk 3 above)
Comment (eddycharly) [inline pkg/apis/v1alpha1/apply.go:10]:
```suggestion
	// DryRun determines whether the file should be applied in dry run mode.
	// +optional
	DryRun bool `json:"dryRun,omitempty"`
```

File: pkg/apis/v1alpha1/create.go (hunk 4 above)
Comment (eddycharly) [inline pkg/apis/v1alpha1/create.go:10]:
```suggestion
	// DryRun determines whether the file should be applied in dry run mode.
	// +optional
	DryRun bool `json:"dryRun,omitempty"`
```

File: pkg/runner/operations/create.go (hunk 2 above)
Comment (shubham-cmyk) [reply pkg/runner/operations/create.go:54]: I think it would be better not a bad idea to have this

**Pull request merged**